Modules/_asynciomodule.c

```diff
@@ -2695,10 +2696,13 @@ task_step_impl(TaskObj *task, PyObject *exc)
         }
     }
 
-    if (result == NULL) {
+    if ((gen_status == PYGEN_RETURN) || result == NULL) {
```

Comment:
it could be just `gen_status != PYGEN_NEXT` (or `gen_status <= 0` if accept my proposition for returned values) if `_PyGen_Send_NameTBD()` was used.